### PR TITLE
📺 Document videos and iframes

### PR DESCRIPTION
**Pull request by rowanc1**
Allow iframes to have captions as well, which turn them into figures.

### CODE REVIEW

File: packages/myst-spec-ext/src/types.ts [hunk 1]
```diff
@@ -64,6 +64,16 @@ export type Image = SpecImage & {
   height?: string;
 };
 
+export type Iframe = {
+  type: 'iframe';
+  label?: string;
+  identifier?: string;
+  src?: string;
```
Comment (fwkoch) [inline packages/myst-spec-ext/src/types.ts:71]: This is functionally required, I think - but maybe there's a reason to not be required in the type.

**Review by fwkoch — APPROVED** [review]
Looks nice!

File: packages/myst-spec-ext/src/types.ts (hunk 1 above)
Comment (rowanc1) [reply packages/myst-spec-ext/src/types.ts:71]: Oh, yes, good point!

Comment (rowanc1) [reply packages/myst-spec-ext/src/types.ts:71]:
```suggestion
  src: string;
```

**Pull request merged**